### ISSUE TITLE
Does not parse distribution information when a package is installed with python setup.py develop (--user)

### ISSUE THREAD

**Issue by cdiener** [Midnighter/dependency-info#15]
#### Problem description

Running the depinfo command line gives bogus output for "~=" requirements.

#### Code Sample

Create a [minimal, complete, verifiable example](https://stackoverflow.com/help/mcve).

```
depinfo "cobra"

Platform Information
--------------------
Linux   5.18.0-1-amd64-x86_64
CPython                3.10.5

Dependency Information
----------------------
appdirs~            missing <---
black                22.3.0
bumpversion         missing
cobra                0.25.0
depinfo~            missing
diskcache~          missing
future               0.18.2
httpx~              missing
importlib-resources   5.6.0
isort               missing
numpy~              missing
optlang~            missing
pandas~             missing
pydantic~           missing
python-libsbml~     missing
rich                 12.2.0
ruamel.yaml~        missing
scipy                 1.8.1
swiglpk               5.0.5
tox                 missing

Build Tools Information
-----------------------
pip        22.1.2
setuptools 62.3.2
wheel      0.37.1
```

COBRAPY was installed with `python3 setup.py develop --user`, not sure if that has something to do with it.

### Context

<details>

```
Platform Information
--------------------
Linux   5.18.0-1-amd64-x86_64
CPython                3.10.5

Dependency Information
----------------------
black               22.3.0
depinfo              2.1.1
importlib-metadata missing
isort              missing
rich                12.2.0
tox                missing

Build Tools Information
-----------------------
pip        22.1.2
setuptools 62.3.2
wheel      0.37.1

```

</details>

**Comment by Midnighter** [comment]
Hi @cdiener,
At the moment I cannot reproduce this. I must admit that I haven't used a `--user` installation in a very long time. I normally still use a virtualenv and then install with `pip install -e .` for development.

**Comment by Midnighter** [comment]
Hmm, actually now I was able to reproduce it. I'm not sure it's worth fixing. I don't see any official guide still suggesting `--user` installations. Do you have a particular reason for that?

**Comment by cdiener** [comment]
Mostly just because pip -e does not allow user installations anymore. Though I would have expected the package to be agnostic of how the dependencies were installed. Might be good to mention that only certain installation paths are supported. I'll try with pipenv.

**Comment by Midnighter** [comment]
I mean, why employ `--user` at all, rather than installing in a virtualenv?

**Comment by cdiener** [comment]
Just preference I guess. Though as far as I know pip defaults to installating with --user if the system path isn't writable. So it's not that uncommon I would say. Or did you mean in particular the develop:user combination?

**Comment by Midnighter** [comment]
Alright, I'll keep it on my radar. When creating

```python
from importlib.metadata import distribution

cobra = distribution("cobra")
cobra.requires
```

looks very different than when installing the package with pip (for reasons unknown to me). In particular, when calling `distribution` recursively on those requirements it seems unable to parse that format which seems like an oversight, really.

**Comment by cdiener** [comment]
Yeah, if is due to installing with `python setup.py develop` it's probably not worth fixing as long as it works with any pip/hatch/poetry-installed packages.

**Comment by cdiener** [comment]
Okay, I can confirm it works in a virtualenv even with `pip install -e`. Feel free to close it.

**Comment by Midnighter** [comment]
Alright, I'll leave it open in case anyone randomly comes by and cares about it but I won't fix it myself for now.
